Source/ASImageNode.mm

```diff
@@ -300,6 +301,17 @@ - (NSObject *)drawParametersForAsyncLayer:(_ASDisplayLayer *)layer
 {
   ASLockScopeSelf();
 
+  if (AS_AVAILABLE_IOS_TVOS(13, 10)) {
+    if (_imageNodeFlags.regenerateFromImageAsset && _image != nil) {
+      _imageNodeFlags.regenerateFromImageAsset = NO;
+      UITraitCollection *tc = [UITraitCollection traitCollectionWithUserInterfaceStyle:_primitiveTraitCollection.userInterfaceStyle];
+      UIImage *updatedImage = [_image.imageAsset imageWithTraitCollection:tc];
+      if ( updatedImage != nil ) {
+        _image = updatedImage;
```

Comment:
So we know this is called from the main thread?

Reply:
Yes the caller of this is main only, with an assertion

Reply:
thanks!